code/_onclick/hud/screen_objects.dm

```diff
@@ -724,6 +724,98 @@
 	screen_loc = ui_living_healthdoll
 	var/filtered = FALSE //so we don't repeatedly create the mask of the mob every update
 
+/atom/movable/screen/healthdoll/human
+	/// Tracks components of our doll, each limb is a separate atom in our vis_contents
+	VAR_PRIVATE/list/atom/movable/screen/limbs
+	/// Lazylist, tracks all body zones that are wounded currently
+	/// Used so we can sync animations should the list be updated
+	VAR_PRIVATE/list/animated_zones
+
+/atom/movable/screen/healthdoll/human/Initialize(mapload, datum/hud/hud_owner)
+	. = ..()
+	limbs = list()
+	for(var/i in GLOB.all_body_zones)
+		var/atom/movable/screen/healthdoll_limb/limb = new(src, null)
+		// layer chest above other limbs, it's the center after all
+		limb.layer = i == BODY_ZONE_CHEST ? layer + 0.05 : layer
+		limbs[i] = limb
+		// why viscontents? why not overlays? - because i want to animate filters
+		vis_contents += limb
+	update_appearance()
+
+/atom/movable/screen/healthdoll/human/Destroy()
+	QDEL_LIST_ASSOC_VAL(limbs)
+	vis_contents.Cut()
+	return ..()
+
+/atom/movable/screen/healthdoll/human/update_icon_state()
+	. = ..()
+	var/mob/living/carbon/human/owner = hud?.mymob
+	if(isnull(owner))
+		return
+	if(owner.stat == DEAD)
+		for(var/limb in limbs)
+			limbs[limb].icon_state = "[limb]DEAD"
+		return
+
+	var/list/current_animated = LAZYLISTDUPLICATE(animated_zones)
+
+	for(var/obj/item/bodypart/body_part as anything in owner.bodyparts)
+		var/icon_key = 0
+		var/part_zone = body_part.body_zone
+
+		var/list/overridable_key = list(icon_key)
+		if(body_part.bodypart_disabled)
+			icon_key = 7
+		else if(owner.stat == DEAD)
+			icon_key = "DEAD"
+		else if(SEND_SIGNAL(body_part, COMSIG_BODYPART_UPDATING_HEALTH_HUD, owner, overridable_key) & OVERRIDE_BODYPART_HEALTH_HUD)
+			icon_key = overridable_key[1] // thanks i hate it
+		else if(!owner.has_status_effect(/datum/status_effect/grouped/screwy_hud/fake_healthy))
+			var/damage = body_part.burn_dam + body_part.brute_dam
+			var/comparison = (body_part.max_damage / 5)
+			if(damage > (comparison * 4))
+				icon_key = 5
```

Comment:
Maybe calculate damage percentage first and then change icons based off that? Also maybe define magic numbers but thats not very important (considering its icon states)